### PR TITLE
[Banner] Expose LayoutMode and add an automatic layout mode.

### PR DESCRIPTION
**Pull request by fluxminds**
An MDCBannerViewLayoutMode variable was used to control the automatically calculated layout style previously. Because users need to manually control the layout style in some case, a `bannerViewLayoutMode` property is exposed to enable this functionality. An `MDCBannerViewLayoutModeAutomatic` enum has been added to preserve the original automatically layout behavior.

Two examples are added in this PR to show the difference between the automatic style mode and the manual style selection.

Screenshots:

Automatic:
Simulator Screen Shot - iPhone 7 - 2019-06-12 at 16 30 47

Manual:
Simulator Screen Shot - iPhone 7 - 2019-06-12 at 16 25 51

### CODE REVIEW

File: components/Banner/src/MDCBannerView.h [hunk 1]
```diff
@@ -16,6 +16,14 @@
 
 #import "MaterialButtons.h"
 
+typedef NS_ENUM(NSInteger, MDCBannerViewLayoutMode) {
```
Comment (jverkoey) [inline components/Banner/src/MDCBannerView.h:22]: Please add documentation for this enumeration.

File: components/Banner/src/MDCBannerView.h [hunk 2]
```diff
@@ -16,6 +16,14 @@
...
+  MDCBannerViewLayoutModeAutomatic,   // Layout is set automatically based on how elements are
+                                      // configured on banner view
```
Comment (jverkoey) [inline components/Banner/src/MDCBannerView.h]: Will the automatic mode ultimately choose one of the modes below behind the scenes? Please document the behavior here.

File: components/Banner/src/MDCBannerView.h [hunk 3]
```diff
@@ -16,6 +16,14 @@
...
+  MDCBannerViewLayoutModeSingleLine,  // All elements on the same line, only supports one button
```
Comment (jverkoey) [inline components/Banner/src/MDCBannerView.h]: “Row” may be more clear throughout these apis and docs due to the conflict of “line” with text label’s “number of lines”. I might expect that a single line layout will also only allow a single line of text, while a single row layout may have multiple lines of text.

File: components/Banner/src/MDCBannerView.h [hunk 4]
```diff
@@ -62,6 +77,6 @@ __attribute__((objc_subclassing_restricted)) @interface MDCBannerView : UIView
 
  The default value is light grey.
  */
-@property(nonatomic, nonnull, strong) UIColor *dividerColor;
+@property(nonatomic, readwrite, strong, nonnull) UIColor *dividerColor;
```
Comment (jverkoey) [inline components/Banner/src/MDCBannerView.h]: Please move this change to a separate PR.

File: components/Banner/src/MDCBannerView.h [hunk 5]
```diff
@@ -16,6 +16,14 @@
 
 #import "MaterialButtons.h"
```
Comment (jverkoey) [inline components/Banner/src/MDCBannerView.h:18]: Had you considered exposing a “layout” object property and providing several pre-built layout objects that can be assigned to it? This would allow clients the flexibility to create entirely custom layouts while still benefiting from the structure of our component + out of the box layouts.

File: components/Banner/src/MDCBannerView.h (hunk 2 above)
Comment (fluxminds) [reply components/Banner/src/MDCBannerView.h]: Yes, it does. I will add the document. The bannerViewLayoutMode, which is public API, always keeps the user preference.

File: components/Banner/src/MDCBannerView.h (hunk 3 above)
Comment (fluxminds) [reply components/Banner/src/MDCBannerView.h]: Good suggestion. Will revise the name here.

File: components/Banner/src/MDCBannerView.h (hunk 5 above)
Comment (fluxminds) [reply components/Banner/src/MDCBannerView.h:18]: Previously, there was a layout class built for this purpose, but removed because of the complexity/extra layer of abstraction. I feel introducing it later when we have more user data and use cases would be an ideal case to make it more flexible and extendable. Since the current API meets the needs and can be easily extended later on because the core concept is the same. How do you feel?

Comment (jverkoey) [reply components/Banner/src/MDCBannerView.h:18]: Makes sense to me! Thanks for the context :)

File: components/Banner/src/MDCBannerView.h (hunk 4 above)
Comment (fluxminds) [reply components/Banner/src/MDCBannerView.h]: Addressed in d457cf307816ca947981521e366f07aa2565dab8.

File: components/Banner/src/MDCBannerView.h (hunk 1 above)
Comment (fluxminds) [reply components/Banner/src/MDCBannerView.h:22]: Addressed in b5d725fdd272ec093f273a732c8948206c06f345.

File: components/Banner/src/MDCBannerView.h (hunk 2 above)
Comment (fluxminds) [reply components/Banner/src/MDCBannerView.h]: Addressed in b5d725fdd272ec093f273a732c8948206c06f345.

File: components/Banner/src/MDCBannerView.h (hunk 3 above)
Comment (fluxminds) [reply components/Banner/src/MDCBannerView.h]: Added a TODO in b5d725fdd272ec093f273a732c8948206c06f345. Will open another PR to address it since a global replace will be needed and it is a naming refactor.

File: components/Banner/src/MDCBannerView.h (hunk 4 above)
Comment (fluxminds) [reply components/Banner/src/MDCBannerView.h]: Addressed in https://github.com/material-components/material-components-ios/pull/7575

File: components/Banner/src/MDCBannerView.h (hunk 1 above)
Comment (jverkoey) [reply components/Banner/src/MDCBannerView.h]: Please make that change in this PR (or if you need to refactor it, please do the refactoring in a separate PR).

**Review by jverkoey — DISMISSED** [review]
Requesting changes for new public APIs.

File: components/Banner/src/MDCBannerView.h (hunk 3 above)
Comment (jverkoey) [reply components/Banner/src/MDCBannerView.h]: Please make that change in this PR because these are new APIs (or if you need to refactor it, please do the refactoring in a separate PR).

Comment (fluxminds) [reply components/Banner/src/MDCBannerView.h]: Addressed in 4ea43103067da9c8aee739e47de6a29ae21b36de.

**Review by jverkoey — APPROVED** [review]
